crates/bevy_animation/src/lib.rs

```diff
@@ -691,217 +740,308 @@ where
         + tangent_in_end * step_duration * (lerp.powi(3) - lerp.powi(2))
 }
 
-#[allow(clippy::too_many_arguments)]
-fn apply_animation(
-    weight: f32,
+fn update_transitions(player: &mut AnimationPlayer, time: &Time) {
+    player.transitions.retain_mut(|animation| {
+        animation.current_weight -= animation.weight_decline_per_sec * time.delta_seconds();
+        animation.current_weight > 0.0
+    });
+}
+
+/// Adds animation support to an app
+#[derive(Default)]
+pub struct AnimationPlugin;
+
+impl Plugin for AnimationPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_asset::<AnimationClip>()
+            .register_asset_reflect::<AnimationClip>()
+            .register_type::<AnimationPlayer>()
+            .add_systems(Update, update_animation_targets)
+            .add_systems(
+                PostUpdate,
+                (animation_player, animate_targets)
+                    .chain()
+                    .before(TransformSystem::TransformPropagate),
+            );
+    }
+}
+
+impl AnimationTarget {
+    // For private use in `update_animation_targets` only.
+    fn new() -> AnimationTarget {
```

Comment:
nit: I think the convention is for `new()` to return `Self`.

```suggestion
    fn new() -> Self {
```

Reply:
We deliberately don't have a style for this: it came up and it wasn't important enough to enforce in CI.

Reply:
There's no CI enforcement maybe but a quick glance at a VSCode search showed tons of `Self` usage. I don't say that's my preference, I just say I prefer if we keep things consistent.

Reply:
Done.